File: Source/Aladdin.CAPI.Java/src/aladdin/capi/Credentials.java
package aladdin.capi;
import java.lang.reflect.*;

///////////////////////////////////////////////////////////////////////////
// Учетные данные аутентификации
///////////////////////////////////////////////////////////////////////////
public abstract class Credentials extends Authentication 
{
    // тип аутентификации
    @SuppressWarnings({"unchecked"}) 
    @Override public final Class<? extends Credentials>[] types()
    {
        // создать массив
        Object array = Array.newInstance(Class.class, 1); 
        
        // указать поддерживаемую аутентификацию
        Array.set(array, 0, getClass()); 
        
        // вернуть созданный массив
        return (Class<? extends Credentials>[])array; 
    }
}

File: Source/Aladdin.CAPI/CAPI/Credentials.cs
using System;

namespace Aladdin.CAPI
{
	///////////////////////////////////////////////////////////////////////////
	// Учетные данные аутентификации
	///////////////////////////////////////////////////////////////////////////
	public abstract class Credentials : Authentication 
	{
		// тип аутентификации
		public override Type[] Types { get { return new Type[] { GetType() }; } }
	}
}
